packages/compiler-cli/src/transformers/api.ts

```diff
@@ -112,6 +112,100 @@ export interface CompilerOptions extends ts.CompilerOptions {
   // This will be true be default in Angular 6.
   fullTemplateTypeCheck?: boolean;
 
+  /**
+   * Whether to check the type of a binding to a directive/component input against the type of the
+   * field on the directive/component.
+   *
+   * For example, if this is `false` then the expression `[input]="expr"` will have `expr` type-
+   * checked, but not the assignment of the resulting type to the `input` property of whichever
+   * directive or component is receiving the binding. If set to `true`, both sides of the assignment
+   * are checked.
+   *
+   * Defaults to `true` if "fullTemplateTypeCheck" is set, `false` otherwise.
+   */
```

Comment:
These should all be marked deprecated as they will be removed in v11

```suggestion
@deprecated
   */
```